### DISCUSSION TITLE
DS1882 support via existing DS3502 driver (I2C driver 67)

### DISCUSSION THREAD

**Discussion by macgeeknz** [arendst/Tasmota#17081] [General]
Hello Tasmota users/devs

I have worked out that Tasmota's driver for the DS3502 digital potentiometer can also be used to drive the DS1882 dual-log potentiometer chip if used in a slightly unusual way.

I am hoping to discover how to get the I2C supported devices list at the following URL updated to include the DS1882 as a supported device:

	https://tasmota.github.io/docs/I2CDEVICES/

A quick outline of what is required of the user to operate it follows.

The DS3502 chip is a single channel pot with 7-bit wiper (wiper positions 0-127). The DS1882 is a dual-channel log pot with 6-bit wipers (2 X wiper positions 0-63).

It turns out that the DS1882 uses the same I2C byte structure for control as does the DS3502, except that if the 7th bit is not set (0) then bits 1-6 are the position setting for the first wiper. If bit 7 is set (1) then bits 1-6 are the position setting for the second wiper.

Therefore to control a DS1882 with Tasmota's existing DS3502 support, one simply needs to send two commands in a row, with 64 added to the right hand channel number (thereby setting the 7th bit).

Updating the right hand channel also switches the left hand channel to zero attenuation (but not vice versa), therefore one must always update the left channel AFTER the right, even if the left channel value has not changed.

For example, to set both wiper positions to 32 (halfway), send the commands:

	wiper1 96
	wiper1 32

Another example sets the left channel to 20 and the right channel to 10 (74-64):

	wiper1 74
	wiper1 10

I hope this helps someone else get their DS1882 based project up and running. I sure appreciate having Tasmota controlling mine :)

Pete

Ps, I was also surprised to find that Maxim let me order 5 of these chips for free with free delivery across the world as 'free samples'! Made for a great shop sound-system attenuator project (wifi volume control).

**Comment by letshin** [comment]
> Ps, I was also surprised to find that Maxim let me order 5 of these chips for free with free delivery across the world as 'free samples'! Made for a great shop sound-system attenuator project (wifi volume control).

Can you talk a bit more about this? Have been looking into this - would you be happy sharing your wifi volume control project?
